### DISCUSSION TITLE
README: How to reference/change rtsp-simple-server.yml inside a docker container

### DISCUSSION THREAD

**Discussion by akennerly** [bluenviron/mediamtx#284] [Questions]
## Which version are you using?

v0.13.1

## Enhancement request

Due to my docker naivete, it took me a while to figure out how to mount my own rtsp-simple-server.yml inside the docker container pulled from dockerhub rather than building my own container.  Others may find this useful.

Add: -v "${PWD}/docker/rtsp-simple-server/rtsp-simple-server.yml:/rtsp-simple-server.yml" to docker run replacing "${PWD}/docker/rtsp-simple-server/rtsp-simple-server.yml" with wherever you want place your rtsp-simple-server.yml on your docker host.

e.g:  docker run --rm -it -p 8554:8554 -v "${PWD}/docker/rtsp-simple-server/rtsp-simple-server.yml:/rtsp-simple-server.yml" aler9/rtsp-simple-server

Since rtsp-simple-server.yml is in the container root I wasn't able to simply mount a replacement config directory from host to container.  docker doesn't let you do that.

**Comment by davidhantson** [comment]
Hi, got the same issue, I'm running all docker containers on the container station of my qnap. It doesn't allow folder sharing of the root directory.
Is it possible to just place the yaml in a config sub folder? That would solve the issue and make this usable ...

Also I will try to use the v parameter with a host directory. But I'm not sure that works out of the box with qnap ...

**Comment by akennerly** [comment]
@davidhantson I tried to map directories at first but found I had to map a specific container file to a host file. A directory in the container set aside for configs and log files would be handy and a bit easier for mapping. Otherwise you need to map each specific file from container to host if they are in the root of the container.

**Comment by dorinclisu** [comment]
You can mount a directory and just start the rtsp server with the non default path inside the container:
```
docker run --rm -it -p 8554:8554 -v "${PWD}/config:/config" aler9/rtsp-simple-server /config/rtsp-simple-server.yml
# assuming that you have rtsp-simple-server.yml inside ${PWD}/config
```

**Comment by davidhantson** [comment]
Thanks a billion for your input you guys, I got it working knowing now I could start rtsp-simple-server with a simple argument /config/rtsp-simple-server.yml . So I used the QNap GUI to do so, no special tricks required or SSH access used. If anyone ever comes across this same issue in QNap Container Station, just PM me and I'll help you out.

**Reply by Rand0mSeed** [reply]
Trying to do the same thing also in QNAP - any chance you could share your Container Station settings. I have limited Docker experience

**Comment by aler9** [comment]
Thanks @dorinclisu
